### ISSUE TITLE
Auth - federatedSignIn - add additional validationData

### ISSUE THREAD

**Issue by pechisworks** [aws-amplify/amplify-js#5522]
**Is your feature request related to a problem? Please describe.**
I would like to add some `validationData` on `Auth.federatedSignIn()` so that i can access this data on some Cognito triggers like `Pre authentication` or `Post authentication`.

**Comment by elorzafe** [comment]
@pechisworks why you cannot use `Auth.signIn(...)` instead of `Auth.federatedSign(..)`, what is your use case?

**Comment by pechisworks** [comment]
Hi @elorzafe.
I want to Sign in with my Facebook or Google-Account. I want to pass some additional data to the `Post authentication` trigger, in my use case I want to pass the IAM `identityId` so I can take over some data from the unauthenticated IAM user. (The IAM user can do some data manipulations, and when he signs up/in, this data should be taken over to the new user)

Is it possible to sign in with a Facebook Account with `Auth.signIn(...)`?

**Comment by alichherawalla** [comment]
@elorzafe  is this possible? 
can we pass validationData to federatedSignIn? 

For my particular use case there is a precursor step that they need to take before they can sign in using facebook, so I pass the result of that step in the validationData to ensure that the user is valid

**Comment by alichherawalla** [comment]
@elorzafe  any update on the above?

**Comment by tobiasriemenschneider** [comment]
+1
Have the same usecase as @alichherawalla.
Would appreciate any update.

**Comment by atakarabt** [comment]
+1
We are facing the same issue, any update?

**Comment by takarabt** [comment]
+1

We are facing the same issue, any update ?

**Comment by DeniferSantiago** [comment]
I'm in the same problem, it is vital for me to pass `validationData` for a function that runs in pre-signup and I also have federated users from Facebook and Google.

**Comment by gabrielmaldi** [comment]
This is key to being able to validate a CAPTCHA in the Pre authentication trigger Lambda.

**Comment by danielvouch** [comment]
Is there any update on this?

**Comment by mohannad-musleh** [comment]
Is there any update on this?

**Comment by atakarabt** [comment]
It's been almost **two years** since this ticket was open, it's really a very important feature to build production-ready sign-in/signup pages using a CAPTCHA. **That will be great to get at least feedback whether you are planning to implement or not**, this will give us the opportunity to think about other solutions like AWS WAF CAPTCHA (not cost-effective) or move to another IDP.

**Comment by SimonHoiberg** [comment]
Is there any update on this?
Why is this being completely ignored?

**Comment by elorzafe** [comment]
@SimonHoiberg I am very sorry I forgot to answer this issue. 

I was looking on [Cognito documentation](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-app-integration.html) and it seems is not possible to send `validationData` or `clientMetadata` when using HostedUI.

I will reach out Cognito team about this problem

**Comment by atakarabt** [comment]
FYI: @elorzafe 

Even for the clients which are not HostedUI based the validationData and clientMetadata not working for federatedSignIn/federatedSignUo

**Comment by hackmajoris** [comment]
Stuck on this issue, too

**Comment by parthNJ** [comment]
Hi, any update on this? Come on man we need a solution for this.

**Comment by abdallahshaban557** [comment]
Hi @parthNJ, unfortunately we are blocked on delivering this feature since we need changes to happen first at the Cognito side. We are in communication with them about these blockers!

**Comment by abdallahshaban557** [comment]
Hi @wookieJ  - unfortunately, not yet. Can you please provide us with the use case that you want to unlock by passing the validationData?

**Comment by wookieJ** [comment]
@abdallahshaban557  We would like to pass fingerprint id to pre-signup lambda using `federatedSignIn()`

**Comment by evc-erik** [comment]
@abdallahshaban557 I'm following this issue as well, not a blocker for me as we are still evaluating Cognito as a solution, but in prototypes I've built this has been an issue. For my use case, I would like the to pass application context that I can rely on in cognito triggers. It basically comes down to tenant information in our use case. We use the same logic for all tenants, but we need to know the tenant identity. For federated logins we would have to infer it from other information since the client has no way to pass it to us.

**Comment by abdallahshaban557** [comment]
This is really helpful context @evc-erik - thank you so much!

**Comment by mtourj** [comment]
Hello. We also need this for our use case.

When users sign up with a Google/Apple/Facebook account with an email associated with an existing Cognito user, I am throwing an error from a pre-signup trigger to ask the client to specify whether to link those accounts or not.

I need to be able to pass a signal to my pre-signup trigger to specify what action the user chooses to take in those cases.

This issue is a blocker for this use cases, and I'll probably have to figure out a hacky workaround to implement this. Not sure what I'll do yet.

This functionality would be very useful for us.

**Comment by turakvlad** [comment]
Hello. Our team needs this as well. I created a new issue explaining our use case [here](https://github.com/aws-amplify/amplify-js/issues/12232).

**Comment by cwomack** [comment]
Just wanted to drop a comment on this issue to say that while there hasn't been any progress to report on this feature request yet, we've marked it to be reviewed with the Cognito team again.  If there's any updates that can be provided, we'll share them as soon as we can!

**Comment by disrael** [comment]
This is a must have in order to pass any extra information to the Post Authentication trigger. We really shouldn't have to justify why that is necessary.

**Comment by taransalh1** [comment]
need this feature for our release

**Comment by hcac10** [comment]
I would say this particular thread is the most active with regard to this issue. Any updates? Also, in the meantime I think a lot of developers could possibly benefit from updating [docs]( https://docs.amplify.aws/react-native/build-a-backend/auth/advanced-workflows/#lambda-triggers) indicating passing metadata when signing in with OAuth redirect is not supported.

**Comment by brianlenz** [comment]
We'd like to be able to pass `ClientMetadata` to the `PreSignup` trigger (via `signInWithRedirect()`) to be able to identify what version of the client is registering an account so that we can give different behavior based on the client. As it is now, we have no way to differentiate clients from each other, and thus we must use a single user experience for all clients (undesirable).

**Comment by alex-cza** [comment]
+1
our team also needs to pass `clientMetadata` to `signInWithRedirect()`and be able to read it in the lambda trigger, any updates on this please?

**Comment by acossta** [comment]
We need this as well

**Comment by mtourj** [comment]
Coming back here to say this issue has become a blocker for another feature. We never got to implement the last feature that this issue blocked, is it possible we can have this functionality so we can a little more control over `signInWithRedirect()` this time?

**Comment by xKelvin** [comment]
We also need this ability. Incredible that this has not been implemented yet as it has been available to CognitoSignUp for years with the `clientMetaData` property.

**Comment by ivadenis** [comment]
we need this as well to store referral ids for new users

**Comment by shiviishiv7** [comment]
is there any update?

**Comment by jwadel** [comment]
The team is so lazy. No reasonable team, in this highly competitive world would have an active user request lingering for over 5 years. I want to believe other teams are definitely not like this.

**Comment by bynov** [comment]
We also need a possibility to send `clientMetadata` during `signInWithRedirect` call. It's an ongoing issue for 5 years, crazy

**Comment by tuan-rowboat** [comment]
We also need this feature !

**Comment by natuan62** [comment]
We also need this feature like signUp !

```
import { signUp } from 'aws-amplify/auth';

await signUp({
  username: 'user@example.com',
  password: 'SecurePassword123!',
  options: {
    clientMetadata: {
      customKey1: 'customValue1',
      customKey2: 'customValue2'
    }
  }
});
```

**Comment by justro** [comment]
Just want to add yet another voice here, with some details about my use case:

We want existing user pool users to be able to associate a social provider login with their existing account. The email address may not be the same, and I have no way of knowing in advance what their social provider ID is for the AdminLinkblahblah call I need to make to associate the social provider with the user pool account.

My idea was to create an API that would generate a random token for the user pool user, and then pass that token into the PreSignUp lambda, so I could find the correct user account and make the AdminLink call. I have added the customState field in the signInWithRedirect, and I can see the custom data is passed into the auth redirect url, but it does not make it into the presignup trigger for me to reference. 

I don't see any fields to use for passing validationData, although the documentation for validationData suggest this is exactly the type of use case it's intended for. Would be great to get an answer to this 5 year old issue.

**Comment by your-diary** [comment]
Last month (2026/01), Cognito added [inbound federation Lambda](https://aws.amazon.com/about-aws/whats-new/2026/01/amazon-cognito-inbound-federation-lambda-trigger/), which is exclusively for federated logins.

This increases the need for and expands possible use-cases of passing client metadata in the context of federated logins.
